### PR TITLE
[bitnami/keycloak] Do not expose https port if tls is not enabled

### PR DESCRIPTION
**Pull request by joancafom**
### Description of the change

Starting Keycloak version 16, WidFly was replaced as the application server by Quarkus. Since then, the HTTPs port is not exposed by default unless specific TLS settings are configured.

This PR aligns the chart with such behaviour, only exposing/showing HTTPs information if TLS is configured.

### Benefits

Chart is consistent with application design

### Possible drawbacks

None, the HTTPs port refused connection.

### Applicable issues

None

### Additional information

https://www.keycloak.org/server/enabletls

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/keycloak/templates/NOTES.txt [hunk 1]
```diff
@@ -24,24 +24,45 @@ To access Keycloak from outside the cluster execute the following commands:
 
 {{- if contains "NodePort" .Values.service.type }}
 
-    export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ include "common.names.fullname" . }})
+    export HTTP_NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ include "common.names.fullname" . }})
```
Comment (fmulero) [inline bitnami/keycloak/templates/NOTES.txt:27]: Little improvement to ensure we are getting the right port:
```suggestion
    export HTTP_NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[?(@.name=='http')].nodePort}" services {{ include "common.names.fullname" . }})
```

File: bitnami/keycloak/templates/NOTES.txt [hunk 2]
```diff
@@ -24,24 +24,45 @@ To access Keycloak from outside the cluster execute the following commands:
...
+    {{- if .Values.auth.tls.enabled }}
+    export HTTPS_NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[1].nodePort}" services {{ include "common.names.fullname" . }})
```
Comment (fmulero) [inline bitnami/keycloak/templates/NOTES.txt:29]: Little improvement to ensure we are getting the right port
```suggestion
    export HTTPS_NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[?(@.name=='https')].nodePort}" services {{ include "common.names.fullname" . }})
```

File: bitnami/keycloak/templates/NOTES.txt [hunk 3]
```diff
@@ -24,24 +24,45 @@ To access Keycloak from outside the cluster execute the following commands:
...
 
 {{- else if contains "LoadBalancer" .Values.service.type }}
 
   NOTE: It may take a few minutes for the LoadBalancer IP to be available.
         You can watch its status by running 'kubectl get --namespace {{ .Release.Namespace }} svc -w {{ include "common.names.fullname" . }}'
 
-    export SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].port}" services {{ include "common.names.fullname" . }})
+    export HTTP_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].port}" services {{ include "common.names.fullname" . }})
+    {{- if .Values.auth.tls.enabled }}
+    export HTTPS_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[1].port}" services {{ include "common.names.fullname" . }})
```
Comment (fmulero) [inline bitnami/keycloak/templates/NOTES.txt:45]: Same here
```suggestion
    export HTTPS_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[?(@.name=='https')].port}" services {{ include "common.names.fullname" . }})
```

File: bitnami/keycloak/templates/NOTES.txt [hunk 4]
```diff
@@ -24,24 +24,45 @@ To access Keycloak from outside the cluster execute the following commands:
...
+    echo "https://${NODE_IP}:${HTTPS_NODE_PORT}/"
+    {{- end }}
 
 {{- else if contains "LoadBalancer" .Values.service.type }}
 
   NOTE: It may take a few minutes for the LoadBalancer IP to be available.
         You can watch its status by running 'kubectl get --namespace {{ .Release.Namespace }} svc -w {{ include "common.names.fullname" . }}'
 
-    export SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].port}" services {{ include "common.names.fullname" . }})
+    export HTTP_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].port}" services {{ include "common.names.fullname" . }})
```
Comment (fmulero) [inline bitnami/keycloak/templates/NOTES.txt:43]: Same here
```suggestion
    export HTTP_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[?(@.name=='http')].port}" services {{ include "common.names.fullname" . }})
```

File: bitnami/keycloak/templates/NOTES.txt [hunk 5]
```diff
@@ -24,24 +24,45 @@ To access Keycloak from outside the cluster execute the following commands:
...
+    {{- end }}
 
 {{- else if contains "ClusterIP" .Values.service.type }}
 
-    export SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].port}" services {{ include "common.names.fullname" . }})
-    kubectl port-forward --namespace {{ .Release.Namespace }} svc/{{ include "common.names.fullname" . }} ${SERVICE_PORT}:${SERVICE_PORT} &
-    echo "http://127.0.0.1:${SERVICE_PORT}/"
+    export HTTP_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].port}" services {{ include "common.names.fullname" . }})
+    {{- if .Values.auth.tls.enabled }}
+    export HTTPS_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[1].port}" services {{ include "common.names.fullname" . }})
```
Comment (fmulero) [inline bitnami/keycloak/templates/NOTES.txt:58]: Same here:
```suggestion
    export HTTPS_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[?(@.name=='https')].port}" services {{ include "common.names.fullname" . }})
```

File: bitnami/keycloak/templates/NOTES.txt [hunk 6]
```diff
@@ -24,24 +24,45 @@ To access Keycloak from outside the cluster execute the following commands:
...
+    {{- if .Values.auth.tls.enabled }}
+    echo "https://${SERVICE_IP}:${HTTPS_SERVICE_PORT}/"
+    {{- end }}
 
 {{- else if contains "ClusterIP" .Values.service.type }}
 
-    export SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].port}" services {{ include "common.names.fullname" . }})
-    kubectl port-forward --namespace {{ .Release.Namespace }} svc/{{ include "common.names.fullname" . }} ${SERVICE_PORT}:${SERVICE_PORT} &
-    echo "http://127.0.0.1:${SERVICE_PORT}/"
+    export HTTP_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].port}" services {{ include "common.names.fullname" . }})
```
Comment (fmulero) [inline bitnami/keycloak/templates/NOTES.txt:56]: Same here

```suggestion
    export HTTP_SERVICE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[?(@.name=='http')].port}" services {{ include "common.names.fullname" . }})
```

**Review by fmulero — CHANGES_REQUESTED** [review]
Great @joancafom !! Thanks a lot.

I left some comments with a little improvement

**Comment by joancafom** [comment]
Thanks for your suggestions @fmulero ! I have already applied them 😁

**Review by fmulero — APPROVED** [review]
LGTM

**Pull request merged**